mesonbuild/compilers/detect.py

```diff
@@ -165,6 +168,8 @@ def detect_static_linker(env: 'Environment', compiler: Compiler) -> StaticLinker
             trials = [defaults['cuda_static_linker']] + default_linkers
         elif compiler.get_argument_syntax() == 'msvc':
             trials = [defaults['vs_static_linker'], defaults['clang_cl_static_linker']]
+        elif is_os2() and env.coredata.get_option(OptionKey('emxomf')):
```

Comment:
```suggestion
        elif env.machines[compiler.for_machine].is_os2() and env.coredata.get_option(OptionKey('emxomf')):
```

`is_os2()` will give the wrong answer in a cross compile. I've been meaning to delete all of those for a while so that we have to use the `Environment.machines` ones, but never seem to get there.

Reply:
Do you mean I should replace all ```is_os2()``` with Environment.machines ones ?

Reply:
the `is_os2()` function is only safe to use if you know that it will always be used for the build machine, so, yes please, use `Environment.machines` if it's available. (There are probably some places we should use that but don't have it available, I'm fine with ignoring those because they're per-exisiting issues)

Reply:
Ok, I'll try.